src/networking.c

```diff
@@ -2688,6 +2689,69 @@ NULL
         } else {
             addReplyLongLong(c,-1);
         }
+    } else if (!strcasecmp(c->argv[1]->ptr,"trackinginfo") && c->argc == 2) {
+        addReplyMapLen(c,4);
+
+        /* Flags */
+        addReplyBulkCString(c,"flags");
+        void *arraylen_ptr = addReplyDeferredLen(c);
+        int numflags = 0;
+        addReplyBulkCString(c,c->flags & CLIENT_TRACKING ? "on" : "off");
+        numflags++;
+        if (c->flags & CLIENT_TRACKING_BCAST) {
+            addReplyBulkCString(c,"bcast");
+            numflags++;
+        }
+        if (c->flags & CLIENT_TRACKING_OPTIN) {
+            addReplyBulkCString(c,"optin");
+            numflags++;
+        }
+        if (c->flags & CLIENT_TRACKING_OPTOUT) {
+            addReplyBulkCString(c,"optout");
+            numflags++;
+        }
+        if (c->flags & CLIENT_TRACKING_NOLOOP) {
+            addReplyBulkCString(c,"noloop");
+            numflags++;
+        }
+        if (c->flags & CLIENT_TRACKING_BROKEN_REDIR) {
+            addReplyBulkCString(c,"broken_redirect");
+            numflags++;
+        }
+        setDeferredSetLen(c,arraylen_ptr,numflags);
+
+        /* Redirect */
+        addReplyBulkCString(c,"redirect");
+        if (c->flags & CLIENT_TRACKING) {
+            addReplyLongLong(c,c->client_tracking_redirection);
+        } else {
+            addReplyLongLong(c,-1);
+        }
+
+        /* Prefixes */
+        addReplyBulkCString(c,"prefixes");
+        if (c->client_tracking_prefixes) {
+            addReplyArrayLen(c,raxSize(c->client_tracking_prefixes));
+            raxIterator ri;
+            raxStart(&ri,c->client_tracking_prefixes);
+            raxSeek(&ri,"^",NULL,0);
+            while(raxNext(&ri)) {
+                addReplyBulkCBuffer(c,ri.key,ri.key_len);
+            }
+            raxStop(&ri);
+        } else {
+            addReplyArrayLen(c,0);
+        }
+
+        /* Caching */
+        addReplyBulkCString(c,"next-key-cached");
+        if (c->flags & CLIENT_TRACKING_OPTIN) {
+            addReplyBulkCString(c,c->flags & CLIENT_TRACKING_CACHING ? "yes" : "no");
+        } else if (c->flags & CLIENT_TRACKING_OPTOUT) {
+            addReplyBulkCString(c,c->flags & CLIENT_TRACKING_CACHING ? "no" : "yes");
+        } else {
+            addReplyBulkCString(c,c->flags & CLIENT_TRACKING ? "yes" : "no");
+        }
```

Comment:
for some reason this still feels odd to me.
maybe it's the new term ("next-key-cached") which i guess isn't yet used anywhere (we probably wanna be consistent with either the commands or existing documentation).

and maybe also the fact that the 'yes' and 'no' values come from so many different sources, it would be hard to reconstruct the state of the configuration. maybe we need 6 unique strings?

or was the whole purpose of this specific field to be easily used?
i.e. the caller knows what's gonna happen on the very next command, rather than understanding the various configurations (if that's the case, i'm not sure it's needed, the caller of the TRACKING command should understand the tracking options).

that said, i'm no CSC expert, i'll ask that others who are will comment.

Reply:
The term we use in the documentation is opt-in caching. My problem is the output is hard to digest without looking at other fields. 

If we don't want to show the flags, maybe this is easy?
"next-command-cached": "Opt-in", "Opt-out", "tracking", "off"

Reply:
i agree with @madolson , the caching flag is only valid for Opt-in Opt-out mode, how about we put this together with the opt-in opt-out tracking flags? maybe we don't need to show it separately here, instead we can show in the flags field?